### PR TITLE
Fix misleading Date constructor documentation (addresses #49350)

### LINKED ISSUE #49350: new Date(year, month) constructor documentation is misleading
# Bug Report

### 🔎 Search Terms

date constructor month 

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about naming conventions

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEBcAsEsGdTqAsgQwJ6gEwAZOYFAB2ApgO6gAiK4xAFDngDSgAsAlEA)

### 💻 Code

```ts
// this is May 2022
new Date(2022, 4)
```

### 🙁 Actual behavior

image

The date constructor documentation gives no indication that the `month` parameter is zero index based (January is month 0, February is month 1). This can cause off-by-one error bugs that are hard to detect.

### 🙂 Expected behavior

I expected the documentation to be clear about the type of value expected in the constructor. For example:
- rename `month` parameter to `monthIndex` (MDN also uses `monthIndex` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax)
- add JSDoc for the constructor explaining the month parameter

### PR DESCRIPTION
**Pull request by jackmorrison12**
Issue number of the reported bug or feature request: https://github.com/microsoft/TypeScript/issues/49350

# Describe your changes

- Added documentation for the Date constructor explaining the purpose of each parameter, wording copied from the UTC method
- Updated the parameter name of `month` to `monthIndex` to align with MDN for Date constructor and UTC method (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax) 

# Testing performed

n/a - exclusively documentation changes and a parameter rename, no way to test these. Updated the baseline changes since some line numbers were updated from the change

Fixes #49350

### CODE REVIEW

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
@jackmorrison12 I'll merge this after you have a chance to merge from main. No need to rebase or anything; I'll squash and merge afterward.

**Comment by mkubilayk** [comment]
@sandersn Thanks for the review - this is now updated.

**Pull request merged**
